xtokens/src/tests.rs

```diff
@@ -431,6 +404,309 @@ fn send_as_sovereign_fails_if_bad_origin() {
 	});
 }
 
+#[test]
+fn para_transact_to_relay_remark_use_sovereign_account() {
+	Relay::execute_with(|| {
+		let _ = RelayBalances::deposit_creating(&para_a_account(), 6030);
+	});
+
+	ParaA::execute_with(|| {
+		parachain_transact_to_relaychian_remark();
+	});
+
+	Relay::execute_with(|| {
+		use relay::{Event, System};
+		assert!(System::events()
+			.iter()
+			.any(|r| matches!(r.event, Event::System(frame_system::Event::Remarked(_, _)))));
+	});
+}
+
+#[test]
+fn relay_transact_to_para_remark_use_default_sovereign_account() {
+	ParaA::execute_with(|| {
+		assert_ok!(ParaTokens::deposit(CurrencyId::R, &DEFAULT, 6030));
+	});
+
+	relaychain_transact_to_parachain_remark(Here, 6030);
+
+	ParaA::execute_with(|| {
+		use para::{Event, System};
+		assert!(System::events()
+			.iter()
+			.any(|r| matches!(r.event, Event::System(frame_system::Event::Remarked(_, _)))));
+	});
+}
+
+#[test]
+fn relay_transact_to_para_remark_use_normal_account() {
+	ParaA::execute_with(|| {
+		assert_ok!(ParaTokens::deposit(CurrencyId::R, &ALICE, 6040));
+		assert_eq!(7040, ParaTokens::free_balance(CurrencyId::R, &ALICE));
```

Comment:
Not needed. We don't test for `deposit` fn here. Same to a lot of following balance equality checks.

Reply:
delete integration test as the testcase is not scope of xTokens pallet. it will be add to acala repo.